jcl/src/java.base/share/classes/java/lang/invoke/DynamicInvokerHandle.java

```diff
@@ -37,6 +37,16 @@
 		this.site = originalHandle.site;
 	}
 
+/*[IF Java15]*/
+	MethodHandle[] getRelatedMHs() {
+		MethodHandle targetMH = site.getTarget();
```

Comment:
All three (`[Mutable|Volatile|Constant]CallSite`) include a `target`. The only difference is that `ConstantCallSite` uses its `target` as the `dynamicInvoker()` where as `[Mutable|Volatile]CallSite` generate a new `dynamicInvoker()` MH, equivalent to an `invokedynamic` instruction, in order to execute the `target`.

Reply:
Right - a ConstantCallSite can be viewed as part of the calling MH chain as there is no option to "break" the chain there.  The other types of CS are different as they have a calling chain, and a callee chain where the callee chain can be changed.

Does the RI treat that as a single thing?  Maybe test with a MCS that targets a DirectMH that throws an exception to validate the assumptions here

Reply:
Still confused about this. Did I properly identify **callee** (`dynamicInvoker()`) and **calling** (`getTarget()`) chains in the below example? We want to verify if `dynamicInvoker()` MH behaves similar to `getTarget()` MH for `MutableCallSite`?

```
MutableCallSite exceptionCS = new MutableCallSite(MethodType.methodType(String.class, Throwable.class));
MethodHandle MH_exception = exceptionCS.dynamicInvoker();

exceptionCS.setTarget(MethodHandles.throwException(String.class, Throwable.class));

try {
        MH_exception.invoke(new SQLException());
} catch (SQLException ex) {
        System.out.println("Caught SQLException callee chain");
}

try {
        exceptionCS.getTarget().invoke(new MalformedURLException());
} catch (MalformedURLException ex) {
        System.out.println("Caught MalformedURLException calling chain");
}
```

Reply:
Not quite.  We want to validate that the RI's equivalent of `hasCheckedException()` follows the MH chain thru the target of a Mutable or Volatile CallSite.

We already know - by spec - how all the CS operate when invoked.  It's the `hasCheckedException()` behaviour that's in question.

So something like:
```java
class C {
public static void m() throws IOException /* a checked exception */ {
}
}

MutableCallSite mcs = new MCS(MethodType.methodType(void.class));
boolean nullTarget = MethodHandles.hasCheckedException(mcs.dynamicInvoker());

MethodHandle mh = Lookup.lookup().findStatic(C.class, "m", methodType(void.class));
mcs.setTarget(mh);
boolean throwingTarget = MethodHandles.hasCheckedException(mcs.dynamicInvoker());

System.out.println("nullTarget = "+nullTaget+"\t throwingTarget = "+throwingTarget);
```

Reply:
RI doesn't find a checked exception in case of `mcs.dynamicInvoker()` but it does in case of `mcs.getTarget()`. So, RI's equivalent of `hasCheckedException()` does not follow the MH chain thru the target of a Mutable or Volatile CallSite. With the below suggestion,  `hasCheckedException()` should match the RI.
```suggestion
		MethodHandle targetMH = site.dynamicInvoker();
```

Reply:
Presumably, we've already processed the dynamicInvoker or we wouldn't be able to get the `site`.  Removing the line should be sufficient

Reply:
`DynamicInvokerHandle.getRelatedMHs` ends up just returning `false` after removing the above line. Change completed.